st2client/st2client/commands/triggerinstance.py

```diff
@@ -0,0 +1,109 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from st2client.commands import resource
+from st2client.commands.noop import NoopCommand
+from st2client.formatters import table
+from st2client.models import TriggerInstance
+from st2client.utils.date import format_isodate
+
+
+class TriggerInstanceBranch(resource.ResourceBranch):
+    def __init__(self, description, app, subparsers, parent_parser=None):
+        super(TriggerInstanceBranch, self).__init__(
+            TriggerInstance, description, app, subparsers,
+            parent_parser=parent_parser,
```

Comment:
You can pass `ready_only=True` argument to the parent class (see #1533 for en example) and only list and get commands will be added.

In any case, yeah, I agree, the parent class code is hacky and I already added some nasty comments there :P